src/preferArrayLiteralRule.ts

```diff
@@ -26,15 +38,26 @@ export class Rule extends Lint.Rules.AbstractRule {
     };
 
     public static GENERICS_FAILURE_STRING: string = 'Replace generic-typed Array with array literal: ';
-    public static CONSTRUCTOR_FAILURE_STRING: string = 'Replace Array constructor with an array literal: ';
-    public static FUNCTION_FAILURE_STRING: string = 'Replace Array function with an array literal: ';
+    public static getReplaceFailureString = (type: InvocationType, nodeText: string) =>
+        `Replace Array ${type} with an array literal: ${nodeText}`;
+    public static getSingleParamFailureString = (type: InvocationType, nodeText: string) =>
+        `To create an array of given length you should use non-negative integer. Otherwise replace Array ${type} with an array literal: ${nodeText}`;
```

Comment:
Would be glad to hear suggestions on better message.

Reply:
I can't think of anything significantly better :smile: other than adding an `a` before `given`.

```
        `To create an array of a given length you should use a non-negative integer. Otherwise replace Array ${type} with an array literal: ${nodeText}`;
```

Also, do these need to print `nodeText`? Seems redundant.

Reply:
I'm really not a fan of adding `nodeText` to an error messages (not only in this rule) because node will be fully underlined in editor when file opened.
However I've added node text because other messages have one. Should I update all messages to drop node text or just this one?

Reply:
Removed `nodeText` from new error message.
There is still question if I should remove it from other messages.

Reply:
Ah, talked offline, I missed that bit in the review here - burn them all! 🔥